src/olympia/devhub/decorators.py

```diff
@@ -56,16 +56,28 @@ def fun():
                 # On read-only requests, ignore disabled so developers can
                 # still view their add-on.
                 if acl.check_addon_ownership(
-                    request, addon, dev=True, ignore_disabled=True
+                    request,
+                    addon,
+                    allow_developer=True,
+                    allow_mozilla_disabled_addon=True,
+                    allow_site_permission=True,
                 ):
                     # Redirect to the submit flow if they're not done.
                     if not submitting and addon.should_redirect_to_submit_flow():
                         return redirect('devhub.submit.details', addon.slug)
                     return fun()
-            # Require an owner or dev for POST requests (if the add-on status
-            # is disabled that check will return False).
+            # Require an owner or deveveloper for POST requests (if the add-on
+            # status is disabled that check will return False). Allow site
+            # permission add-ons, an extra check will be made for pages that
+            # don't make sense if the add-on is a site permission add-on
+            # through the @readonly_if_site_permission decorator below.
```

Comment:
I don't like the way `@dev_required` now defaults to allowing writes to site_permission add-ons, and you have to remember to add a second decorator to make it safe - the ideal situation is it defaults to preventing writes to those add-ons.

IMO add a `allow_site_permission=False` kwarg to the decorator and pass that here.